components/b2b/AccountHierarchy/AccountHierarchyForm/AccountHierarchyForm.tsx

```diff
@@ -0,0 +1,238 @@
+import { useEffect, useState } from 'react'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import { LoadingButton } from '@mui/lab'
+import { Button, FormControl, MenuItem, Stack } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+import { useForm, Controller } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { AccountHierarchyFormStyles } from './AccountHierarchyForm.styles'
+import { KiboSelect, KiboTextBox } from '@/components/common'
+import { CreateCustomerB2bAccountParams } from '@/lib/types'
+
+import { CustomerAccount } from '@/lib/gql/types'
+
+interface AccountHierarchyFormProps {
+  accounts?: CustomerAccount[]
+  isAddingAccountToChild: boolean
+  onSave: (data: CreateCustomerB2bAccountParams) => void
+  onClose: () => void
+}
+
+const useAccountHierarchySchema = () => {
+  const { t } = useTranslation('common')
+  return yup.object({
+    parentAccount: yup.string().required(t('this-field-is-required')),
+    companyOrOrganization: yup.string().required(t('this-field-is-required')),
+    firstName: yup.string().required(t('this-field-is-required')),
+    lastName: yup.string().required(t('this-field-is-required')),
+    emailAddress: yup.string().required(t('this-field-is-required')),
+  })
+}
+
+const AccountHierarchyForm = (props: AccountHierarchyFormProps) => {
+  const { accounts, isAddingAccountToChild, onSave, onClose } = props
+
+  const [isLoading, setLoading] = useState<boolean>(false)
+  const [selectedParentAccount, setSelectedParentAccount] = useState<CustomerAccount>()
+
+  const { t } = useTranslation()
+  const accountHierarchySchema = useAccountHierarchySchema()
+
+  const {
+    formState: { errors, isValid },
+    control,
+    getValues,
+    setValue,
+    handleSubmit,
+  } = useForm({
+    defaultValues: {
+      parentAccount: '',
+      companyOrOrganization: '',
+      firstName: '',
+      lastName: '',
+      emailAddress: '',
+      taxId: '',
+    },
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    criteriaMode: 'firstError',
+    resolver: yupResolver(accountHierarchySchema),
+    shouldFocusError: true,
+  })
+
+  useEffect(() => {
+    const hasMultipleAccounts = accounts?.length && accounts.length > 1
+    if (!hasMultipleAccounts) {
+      setValue('parentAccount', accounts?.[0]?.companyOrOrganization as string)
+      setSelectedParentAccount(accounts?.[0])
+    }
+  }, [accounts])
+
+  const onSubmit = () => {
+    if (isLoading || !isValid) return
+    setLoading(true)
+    const formValues = getValues()
+    onSave({
+      ...formValues,
+      parentAccount: selectedParentAccount,
+    })
+  }
```

Comment:
After onSave where we are setting setLoading(false)?

Reply:
onSave method doesn't return the promise. 
setLoading(false) stops the loader immediately. 
By not using setLoading(false) loader is visible until the form is submitted. 
After successful response dialog closes so not required to set isLoading false.